src/connectors/utils/mime.rs

```diff
@@ -63,17 +64,35 @@ impl MimeCodecMap {
     }
 
     /// constructs this map while overriding the mapping from mime-type to codec by `custom_codecs`
-    pub fn with_overwrites(custom_codecs: &HashMap<String, String>) -> Self {
-        let mut base = Self::new();
-        for (mime, codec_name) in custom_codecs {
-            base.by_mime.insert(mime.clone(), codec_name.clone());
+    pub(crate) fn from_custom(custom_codecs: HashMap<String, String>) -> Self {
+        // We use the existing codec -> mimetype mapping as default to ensure that each
+        // codec at least maps to a mime type
+        let mut by_codec = CODEC_TO_MIME_TYPES
+            .into_iter()
+            .map(|(k, v)| (k.to_string(), v.to_string()))
+            .collect::<HashMap<String, String>>();
+
+        // After that we overwrite / extend the mapping with the definition given
+        // by the user
+        for (mime, codec_name) in &custom_codecs {
+            if mime != "*/*" {
+                by_codec.insert(mime.to_string(), codec_name.to_string());
```

Comment:
this is inserted the wrong way (or into the wrong map).

Reply:
good catch thank you! That was me breaking things in refactoring :(

Reply:
fixed as 👍 
```rust
        for (mime, codec_name) in &custom_codecs {
            if mime != "*/*" {
                by_codec.insert(codec_name.clone(), mime.clone());
            }
        }
```
I'll push it with other feedback / the next round